### PR TITLE
Show rate ped

### PR DESCRIPTION
**Pull request by jacob-stein1**
# Description

## Type of Change

- [x] Possible breaking change (Change in feature that may cause existing features to not work correctly, e.g. change in
  schema)
- [ ] Non breaking change (Bug fix, New feature)

## What type of PR is this?

- [x] bug fix
- [ ] cleanup
- [x] documentation
- [ ] feature

## What this PR does / Why we need it:

```docs
Adds a max and min value for show-rate, and also changes API key
```

## Which issue(s) this PR fixes, if:

Fixes # 134, 164, 129, 127

#### Additional documentation e.g., usage docs, testing docs etc.:

```docs

```

## Checklist

- [x] Changes have been tested
- [x] All automated tests pass
- [x] Relevant Tags and Milestones have been added

### CODE REVIEW

File: src/main/java/seedu/duke/commands/ShowRateCommand.java [hunk 1]
```diff
@@ -65,18 +67,21 @@ public String getRateString (Forex instance, BigDecimal amt) {
...
             return new BigDecimal(1);
         }
-        float val = Float.valueOf(args[3]); // Potential IllegalArgumentException
-        if (val < 0) {
-            throw new InvalidNumberException(); // For negative inputs
+        if (!args[3].matches("[0-9\\.-]+")) {
+            throw new IllegalArgumentException();
         }
-        return new BigDecimal(val);
+        BigDecimal val = new BigDecimal(args[3]); // Potential NumberFormatException
```
Comment (Arif-Khalid) [inline src/main/java/seedu/duke/commands/ShowRateCommand.java:80]: Should the potential NumberFormatException as commented be unhandled? 
I suggest instead of checking with String.matches, handling the NumberFormatException accomplishes the same thing in a much clearer manner.

File: src/main/java/seedu/duke/constants/Message.java [hunk 2]
```diff
@@ -34,13 +34,14 @@ public enum Message {
...
-    LOADING_EXCHANGE_RATES("Initializing live exchange rates. Please wait 5 seconds..."),
+    LOADING_EXCHANGE_RATES("Initializing live exchange rates. Please wait for a confirmation message..."),
     SUCCESSFUL_CREATE_ACCOUNT_COMMAND("You have successfully added the %s account\n"),
     SHOW_ALL_TRANSACTIONS_HEADER("Below are all your transactions in reverse chronological order:"),
     SHOW_TRANSACTIONS_OF_DESC_HEADER("Below are all your transactions with the description "),
     SHOW_TRANSACTIONS_OF_DATE_HEADER("Below are all your transactions with the date "),
     SHOW_TRANSACTIONS_OF_CURRENCY_HEADER("Below are all your transactions with currency "),
-    SHOW_TRANSACTIONS_OF_MONTH_HEADER("Below are all your transactions during the month of ");
+    SHOW_TRANSACTIONS_OF_MONTH_HEADER("Below are all your transactions during the month of "),
+    API_INITIALIZED("Real-time exchange rates have been set! You are free to use MoneyMoover!");
```
Comment (Arif-Khalid) [inline src/main/java/seedu/duke/constants/Message.java:44]: Good message that informs the users when the application is ready

File: src/main/java/seedu/duke/commands/ShowRateCommand.java (hunk 1 above)
Comment (Arif-Khalid) [reply src/main/java/seedu/duke/commands/ShowRateCommand.java:80]: Perhaps I am misunderstanding the significance of the String.matches call in relation to the issues this PR solves, please clarify if so.

File: src/test/java/seedu/duke/commands/ShowRateCommandTest.java [hunk 3]
```diff
@@ -81,4 +81,24 @@ public void testParseAmount_negativeValue_shouldThrowInvalidNumberException () {
...
+
+    @Test
+    public void testParseAmount_smallValue_shouldThrowInvalidNumberException () {
+        try {
+            ShowRateCommand cmd = new ShowRateCommand("show-rate THB SGD 0.000001");
+            assertThrows(InvalidNumberException.class, cmd::parseAmount);
+        } catch (Exception e) {
+            fail();
+        }
+    }
```
Comment (Arif-Khalid) [inline src/test/java/seedu/duke/commands/ShowRateCommandTest.java:103]: Great use of Boundary Value Analysis in your test case design

**Review by Arif-Khalid — CHANGES_REQUESTED** [review]
Good bug fix to the show-rate command, however implementation could be improved or requires more clarity to be understood
